### ISSUE TITLE
Possible bug in label vs core counts

### ISSUE THREAD

**Issue by priamai** [d-chambers/dbscan1d#7]
Hi,
found an issue when counting labels with core points.
If you try this:
```

from sklearn.datasets import make_blobs

from dbscan1d.core import DBSCAN1D

import random
random.seed(0)

for x in range(1,100):
  # make blobs to test clustering
  X,y = make_blobs(1_000_000, centers=2, n_features=1)

  # init dbscan object
  dbs = DBSCAN1D(eps=.5, min_samples=4)

  # get labels for each point
  labels = dbs.fit_predict(X)
  core_pts = dbs.core_sample_indices_

  core_size = core_pts[0].size
  label_size = np.where(labels >=0)[0].size

  if core_size != label_size:
    print('Total points %d' % len(X))
    print('Cluster ID: %s' % np.unique(labels))
    print('Total noise %s' % np.where(labels <0)[0].size)
    print('Total core %s' % np.where(labels >=0)[0].size)
    print('Total core points %d' % core_pts[0].size)

```
Very often you will find this situation:

```
Total points 1000000
Cluster ID: [-1  0  1]
Total noise 1
Total core 999999
Total core points 999998

```
The labels do not match the core point counts.

**Comment by d-chambers** [comment]
Hi @priamai,

My apologies but I am just now getting back to this. If you are still interested, can you elaborate a little more? There could be the case where there are a few points on the end of the cluster that aren't core points no?

For example:

labels = [0, 0, 0, 0,  -1, 1, 1, 1, -1]

labels[0], labels[3], labels[5], labels[7] could all belong to clusters without being core points. There would be a problem if the non-core points (included in a cluster) aren't occurring on the edges of the cluster.

**Comment by d-chambers** [comment]
So I did a bit more testing on this, and I found dbscan1d is performing the same as dbscan from sklearn. I did have to reduce the number of samples though...

```python
from sklearn.datasets import make_blobs

from dbscan1d.core import DBSCAN1D

import random
random.seed(0)

def clusters_equivalent(labels1, labels2):
    """
    Return True if two arrays defining cluster structures are equivalent.

    Accounts for each array having a different name for each cluster.
    """
    array = np.array([labels1, labels2]).T
    # get the unique pairings of data
    unique_pairs = np.unique(array, axis=0)
    # Ensure each number is paired with exactly one other number
    left = np.unique(unique_pairs[:, 0])
    right = np.unique(unique_pairs[:, 1])
    return len(left) == len(right) == len(unique_pairs)

for x in range(1, 100):
    # make blobs to test clustering
    X, y = make_blobs(10_000, centers=2, n_features=1)

    # init dbscan object
    dbs_1 = DBSCAN1D(eps=.5, min_samples=4)
    dbs_2 = DBSCAN(eps=0.5, min_samples=4)

    # get labels for each point
    labels_1 = dbs_1.fit_predict(X)
    labels_2 = dbs_2.fit_predict(X)
    assert clusters_equivalent(labels_1, labels_2)

    core_pts_1 = dbs_1.core_sample_indices_
    core_pts_2 = dbs_2.core_sample_indices_
    assert np.all(core_pts_1 == core_pts_2)
```

Feel free to reopen if you find this is not the case or have something to add.
